Example/ViewController.swift

```diff
@@ -125,9 +125,11 @@ class ViewController: UIViewController {
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-
-        self.mapView = NavigationMapView(frame: view.bounds)
-
+        
+        if mapView == nil {
+            mapView = NavigationMapView(frame: view.bounds)
```

Comment:
does it look like indentation is off? does xcode autoformating on our behalf?

Reply:
i beg your pardon @JThramer but couldn't be the same achieved with `lazy` property? which could also help with DRY on line 510?

Reply:
Regarding indentation, Xcode does automatically indent code. We stick to the default of four spaces per indent, which appears to be correct here. We’ve avoided adding a style linter because stuff like whitespace rarely comes up in review. Occasionally indentation gets flubbed and I’ll nitpick about it if it happens to hurt readability.

The actual fix could bear some explanation in inline comments, given that this is an example application. The application mostly follows [this guide](https://docs.mapbox.com/ios/api/navigation/0.36.0/reduce-memory-usage.html) for reducing memory usage in a multi-scene UI. The guide assumes that the map view is set up in the first place.

Previously, this view controller assumed that the default modal presentation style would cover up the whole screen, causing `viewDidAppear(_:)` to get called when exiting navigation. In iOS 13, the default changed to a card-like presentation, which meant the root view would never reappear because it always remained visible. This PR forces the full screen presentation, which was previously the default, and just in case, it also explicitly restores the map view when canceling navigation. As it happens, full screen presentation means that canceling navigation causes the root view to reappear, so both initializations could occur, hence the nil check.

A lazy property wouldn’t work because we do need the map view to be torn down and restored multiple times as the user enters, exits, and reenters navigation. Also, the `mapView` property not only holds the map view but also installs it in the view hierarchy. So setting this property has a side effect. Perhaps this is too much cleverness for an ostensible example application and explicit teardown and restoration would be more appropriate.

Reply:
Would have been nice to see this in PR description:
This PR forces the full screen presentation, which was previously the default, and just in case, it also explicitly restores the map view when canceling navigation. As it happens, full screen presentation means that canceling navigation causes the root view to reappear, so both initializations could occur, hence the nil check.

Reply:
Good point. I'll try and be more explicit next time.